etcdserver/auth/auth_test.go

```diff
@@ -71,16 +86,10 @@ func TestMergeUser(t *testing.T) {
 			User{User: "foo", Roles: []string{"role1", "role2"}},
 			false,
 		},
-		{
-			User{User: "foo"},
-			User{User: "foo", Password: "$2a$10$aUPOdbOGNawaVSusg3g2wuC3AH6XxIr9/Ms4VgDvzrAVOJPYzZILa"},
-			User{User: "foo", Roles: []string{}, Password: "$2a$10$aUPOdbOGNawaVSusg3g2wuC3AH6XxIr9/Ms4VgDvzrAVOJPYzZILa"},
-			false,
-		},
```

Comment:
We need to have a test case to cover your new change. We need to ensure the empty password will not overwrite the previous password. 

Can we add
```go
{
    User{User: "foo"},
    User{User: "foo", "ps"},
    User{User: "foo", "ps"},
    false,
}
```
?